tidy3d/components/base.py

```diff
@@ -75,9 +75,89 @@ def __hash__(self) -> int:
 
     def __init__(self, **kwargs):
         """Init method, includes post-init validators."""
+
+        # initialize warning tracing and appoint ourselves as root who strated it
```

Comment:
in the future, would be best to separate this into a couple methods, so the init method would look something like
```py
    def __init__(self, **kwargs):
        """Init method, includes post-init validators."""
        self._initialize_log_cache()
        super().__init__(**kwargs)
        self._post_init_validators()
       self._consolidate_log_cache()
```

Reply:
yeah, I just left it all exposed for now, if this is something we want to move forward with, then I'll definitely wrap it all into separate functions